flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/connectors/DynamicSourceUtils.java

```diff
@@ -483,8 +483,11 @@ private static void validateScanSourceForStreaming(
             ChangelogMode changelogMode,
             ReadableConfig config) {
         // sanity check for produced ChangelogMode
-        final boolean hasUpdateBefore = changelogMode.contains(RowKind.UPDATE_BEFORE);
-        final boolean hasUpdateAfter = changelogMode.contains(RowKind.UPDATE_AFTER);
+        final boolean hasChangelogMode = changelogMode != null;
```

Comment:
Checking why `changlegMode == null` of the failed case?

Reply:
because the mocked table (`TestDynamicTableFactory.DynamicTableSourceMock`) used in test case `DynamicTableSourceSpecSerdeTest` returns null changelog mode(may also be the case in user implemented connectors), so the protection was added